test/Test/MockDockerEngineApi.hs

```diff
@@ -0,0 +1,179 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Test.MockDockerEngineApi (
```

Comment:
A smaller version of `Test.MockApi`. Intentionally going with a copy instead of refactoring MockApi code. I have added this in the final container scanning cleanup work item.